src/style/mod.rs

```diff
@@ -94,6 +94,46 @@ impl Default for Position {
     }
 }
 
+/// How children overflowing their container should affect layout
+///
+/// In CSS the primary effect of this property is to control whether contents of a parent container that overflow that container should
+/// be displayed anyway, be clipped, or trigger the container to become a scroll container. However it also has secondary effects on layout,
+/// the main ones being:
+///
+///   - The automatic minimum size Flexbox/CSS Grid items with non-`Visible` overflow is `0` rather than being content based
+///
+/// In Taffy, we only implement the layout related secondary effects as we are not concerned with drawing/painting.
+///
+/// <https://developer.mozilla.org/en-US/docs/Web/CSS/overflow>
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Default)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum Overflow {
+    /// The automatic minimum size of this node as a flexbox/grid item should be based on the size of it's content.
+    #[default]
+    Visible,
+    /// The automatic minimum size of this node as a flexbox/grid item should be `0`.
+    Hidden,
+}
+
+impl Overflow {
+    /// Returns true for overflow modes that contain their contents (`Overflow::Hidden`, `Overflow::Scroll`, `Overflow::Auto`)
+    /// or else false for overflow modes that allow their contains to spill (`Overflow::Visible`).
+    #[inline(always)]
+    pub(crate) fn is_scroll_container(self) -> bool {
```

Comment:
Perhaps the naming is a bit misleading here as there is no scrolling going on in `Overflow::Hidden`, is there?
However, I also can't think of a better name right now.
`is_containing_container` doesn't sound so good :D

Reply:
I agree that it's a bit confusing, but according to the CSS spec there actually is scrolling with `overflow: hidden`! There are no scrollbars (or scrollwheel support), but in browsers there is support for programatically scrolling an `overflow: hidden` element. There is also a newish `overflow: clip` value that does clipping without being a scroll container.

I did have this method called `contains_contents` but I ended up switching it. This was partly to align with the terminology the spec uses and partly in anticipation of possible future `overflow: clip` support (intuitively `contains_contents` ought to return true for `overflow: clip` nodes, but it shouldn't for how I'm currently using it).

Reply:
Interesting, good to know!